Backend/Data/Struct/OrderedMoveList.cs

```diff
@@ -23,26 +23,27 @@ namespace Backend.Data.Struct;
     };
     
     private readonly Span<OrderedMoveEntry> Internal;
+
+    private readonly OrderedMoveEntry KillerMoveOne;
+    private readonly OrderedMoveEntry KillerMoveTwo;
     
-    [MethodImpl(MethodImplOptions.AggressiveOptimization)]
-    private static int ScoreMove(
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    private int ScoreMove(
         Board board, 
         ref OrderedMoveEntry move,
         SearchedMove tableMove
         )
     {
-        if (move == tableMove) {
-            return PRIORITY - 1;
-        }
+        if (move == tableMove) return PRIORITY - 1;
         
-        if (move.Promotion != Promotion.None) {
-            return PRIORITY - 8 + (int)move.Promotion;
-        }
+        if (move.Promotion != Promotion.None) return PRIORITY - 8 + (int)move.Promotion;
 
         Piece to = board.At(move.To).Item1;
-        if (to != Piece.Empty) {
-            return MvvLva(board.At(move.From).Item1, to) * 1000;
-        }
+        if (to != Piece.Empty) return MvvLva(board.At(move.From).Item1, to) * 1000;
+
+        if (move == KillerMoveOne) return 9000;
+        // ReSharper disable once ConvertIfStatementToReturnStatement
+        if (move == KillerMoveTwo) return 8000;
```

Comment:
In general, "else if" style can be faster because in the series of ifs, every condition is checked one after the other; in an "else if" chain, once one condition is matched, the rest are bypassed. Would this be slight improvement for lines 44-46?

Reply:
I completely understand what you mean. However, in this case, we are simply returning. So when one match happens, we just return rather than allowing the further continuation of that method.